cloudevents/working-drafts/dds-protocol-binding.md

```diff
@@ -0,0 +1,265 @@
+# Data Distribution Service Protocol Binding for CloudEvents - Version 1.0.3-wip
+
+## Abstract
+
+The [Data Distribution Service(DDS)][dds] Protocol Binding for CloudEvents
+defines how events are mapped to [DDS messages][dds-message-format].
+
+## Table of Contents
+
+1. [Introduction](#1-introduction)
+
+- 1.1. [Conformance](#11-conformance)
+- 1.2. [Relation to DDS](#12-relation-to-dds)
+- 1.3. [Content Modes](#13-content-modes)
+- 1.4. [Event Formats](#14-event-formats)
+- 1.5. [Security](#15-security)
+
+2. [Use of CloudEvents Attributes](#2-use-of-cloudevents-attributes)
+
+- 2.1. [data](#21-data)
+
+3. [DDS Message Mapping](#3-dds-message-mapping)
+
+- 3.1. [Keys](#31-keys)
+- 3.2. [Binary Content Mode](#32-binary-content-mode)
+- 3.3. [Structured Content Mode](#33-structured-content-mode)
+
+4. [References](#4-references)
+
+## 1. Introduction
+
+[CloudEvents][ce] is a standardized and protocol-agnostic definition of the
+structure and metadata description of events. This specification defines how the
+elements defined in the CloudEvents specification are implemented in the DDS protocol
+and transmitted as [DDS messages][dds-message-format].
+
+### 1.1. Conformance
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD",
+"SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be
+interpreted as described in [RFC2119][rfc2119].
+
+### 1.2. Relation to DDS
+
+This specification does not prescribe rules constraining transfer or settlement
+of event messages with DDS; it solely defines how CloudEvents are expressed in
+the DDS protocol as [DDS messages][dds-message-format].
+
+The DDS protocol is an [Object Management Group (OMG)][omg] middleware standard
+for efficient and real-time data exchange in distributed systems.
+It enables communication using a publish-subscribe model and
+operates in conjunction with the [Real-Time Publish Subscribe (RTPS)][rtps]
+wire protocol.
+DDS ensures reliable data delivery for latency-sensitive payloads,
+supporting Quality of Service (QoS) parameters like reliability, lifespan
+and partitioning. DDS is most commonly used for mission-critical applications
+in industries such as aerospace, healthcare, and industrial automation.
+
+This binding specification defines how attributes and data of a CloudEvent is
+mapped to a DDS message format.
+
+### 1.3. Content Modes
+
+The CloudEvents specification defines three content modes for transferring
+events: _structured_, _binary_ and _batch_. 
+
+Every compliant implementation of the CloudEvent specification SHOULD support
+both structured and binary modes. The DDS protocol binding supports both these modes.
+
+The DDS protocol binding does not currently support the _batch_ content mode.
+
+In the _structured_ content mode, event metadata attributes and event data are
+placed into the DDS message using an [event format](#14-event-formats).
```

Comment:
Just curious, the next paragraph mentions the `data` section but this paragraph does not, is it correct to assume that in structured mode the entire CE goes into the `data` section?

Reply:
That is correct. In structured mode, the entire CE message would go into the data section.

Reply:
I updated the wording a bit between the two sections to make the difference between structured and binary modes a bit more clear.